src/Core/SecretsManager/Commands/EnableAccessSecretsManager/EnableAccessSecretsManagerCommand.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.Exceptions;
+using Bit.Core.Repositories;
+using Bit.Core.SecretsManager.Commands.EnableAccessSecretsManager.Interfaces;
+
+namespace Bit.Core.SecretsManager.Commands.EnableAccessSecretsManager;
+
+public class EnableAccessSecretsManagerCommand : IEnableAccessSecretsManagerCommand
+{
+    private readonly IOrganizationUserRepository _organizationUserRepository;
+
+    public EnableAccessSecretsManagerCommand(IOrganizationUserRepository organizationUserRepository)
+    {
+        _organizationUserRepository = organizationUserRepository;
+    }
+
+    public async Task<List<(OrganizationUser, string)>> EnableUsersAsync(Guid organizationId,
+        ICollection<OrganizationUser> organizationUsers)
+    {
+        var filteredUsersByOrg = organizationUsers.Where(ou => ou.OrganizationId == organizationId)
+            .ToList();
```

Comment:
I'm not 100% sure on this, but I wonder if this validation should be done outside the command. Do we tell the command, "enable SM for these orgUsers for this organization"? Or should we just tell it, "enable SM for these orgUsers"? Specifying the organization seems redundant; should the command just trust what's passed into it?

I prefer the latter, but I don't know if there's a right or wrong answer, so it's your call. Just thought I'd mention it for discussion.

Reply:
I think it makes sense to take that out of the command. It really is a query concern when thinking about the CQRS pattern. 

Resolved in https://github.com/bitwarden/server/pull/3020/commits/015c68aa31ccdeb0b246f7d3a903eb1312174b3d